test/test-resourcelimits.ts

```diff
@@ -0,0 +1,24 @@
+import Piscina from '..';
+import { test } from 'tap';
+import { resolve } from 'path';
+
+test('resourceLimits causes task to reject', async ({ is, match }) => {
+  const worker = new Piscina({
+    filename: resolve(__dirname, 'fixtures/resource-limits.js'),
+    resourceLimits: {
+      maxOldGenerationSizeMb: 16,
+      maxYoungGenerationSizeMb: 4,
+      codeRangeSizeMb: 16
+    }
+  });
+  const limits : any = worker.options.resourceLimits;
+  is(limits.maxOldGenerationSizeMb, 16);
+  is(limits.maxYoungGenerationSizeMb, 4);
+  is(limits.codeRangeSizeMb, 16);
+  try {
+    await worker.runTask(null);
+  } catch (err) {
+    match(err.message,
+      /Worker terminated due to reaching memory limit: JS heap out of memory/);
+  }
```

Comment:
I think `rejects` should work here – the issue seems to be that because this is run under `tap`, typescript compilation support is also loaded for the Worker thread, which brings the heap over the limits set here on macOS (but no idea why specifically there)